### PR TITLE
Replace Jackson usage with azure-json

### PR DESCRIPTION
**Pull request by alzimmermsft**
Replaces the usage of Jackson with `azure-json`.

### CODE REVIEW

File: customization-base/src/main/java/com/azure/autorest/customization/implementation/ls/EclipseLanguageClient.java [hunk 1]
```diff
@@ -215,14 +209,7 @@ public void close() {
...
     private static <T> T sendRequest(Connection connection, String method, Object param, Type responseType) {
-        try {
-            String response = OBJECT_MAPPER.writeValueAsString(connection.requestWithSerializedObject(
-                OBJECT_MAPPER.constructType(JsonNode.class), method, GSON.toJson(param)));
-
-            return GSON.fromJson(response, responseType);
-        } catch (IOException e) {
-            throw new UncheckedIOException(e);
-        }
+        return GSON.fromJson(connection.requestWithSerializedObject(method, GSON.toJson(param)), responseType);
```
Comment (weidongxu-microsoft) [inline customization-base/src/main/java/com/azure/autorest/customization/implementation/ls/EclipseLanguageClient.java:212]: Are we replacing it wit gson?

File: javagen/src/main/java/com/azure/autorest/util/TemplateUtil.java [hunk 2]
```diff
@@ -81,9 +79,12 @@ public class TemplateUtil {
      * @return the JSON string
      */
     public static String prettyPrintToJson(Object jsonObject) {
```
Comment (weidongxu-microsoft) [inline javagen/src/main/java/com/azure/autorest/util/TemplateUtil.java:81]: I guess this is no longer "pretty print".

As long as SDK CI won't report anything, I am fine with the formatting of json file.

**Review by weidongxu-microsoft — APPROVED** [review]

**Review by XiaofeiCao — APPROVED** [review]

File: customization-base/README.md [hunk 3]
```diff
@@ -181,14 +181,14 @@ with customization
...
     ClassCustomization foo = models.getClass("Foo");
-    foo.addAnnotation("JsonClassDescription(\"Foo class\")");
+    foo.addAnnotation("Deprecated");
 }
 ```
 
 will generate
 
 ```java readme-sample-add-class-annotation-result
-@JsonClassDescription("Foo class")
```
Comment (srnagar) [inline customization-base/README.md:191]: Why is `@JsonClassDescription` replaced with `@Deprecated`?

File: customization-base/src/main/java/com/azure/autorest/customization/implementation/ls/EclipseLanguageClient.java (hunk 1 above)
Comment (srnagar) [reply customization-base/src/main/java/com/azure/autorest/customization/implementation/ls/EclipseLanguageClient.java:212]: Why do we want to use GSON here?

File: extension-base/src/main/java/com/azure/autorest/extension/base/jsonrpc/CallerResponse.java [hunk 4]
```diff
@@ -1,71 +0,0 @@
...
-
-import java.util.concurrent.CompletableFuture;
-import java.util.function.Function;
-
-/**
- * Represents a response from a remote procedure call.
- *
- * @param <T> The type of the result of the remote procedure call.
- */
-public class CallerResponse<T> extends CompletableFuture<T> {
```
Comment (srnagar) [inline extension-base/src/main/java/com/azure/autorest/extension/base/jsonrpc/CallerResponse.java:16]: Why is this type removed? Was it never used?

File: fluentgen/pom.xml [hunk 5]
```diff
@@ -56,6 +56,18 @@
...
+      <artifactId>jackson-databind</artifactId>
+      <version>2.13.5</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>com.fasterxml.jackson.datatype</groupId>
+      <artifactId>jackson-datatype-jsr310</artifactId>
+      <version>2.13.5</version>
+      <scope>test</scope>
+    </dependency>
```
Comment (srnagar) [inline fluentgen/pom.xml:70]: Why do we add jackson dependency here? Should we use azure-json for tests as well and not require this dependency?

File: customization-base/src/main/java/com/azure/autorest/customization/implementation/ls/EclipseLanguageClient.java (hunk 1 above)
Comment (alzimmermsft) [reply customization-base/src/main/java/com/azure/autorest/customization/implementation/ls/EclipseLanguageClient.java:212]: The types used by the Eclipse dependencies use GSON

File: customization-base/README.md (hunk 3 above)
Comment (alzimmermsft) [reply customization-base/README.md:191]: This was to get Jackson removed as a dependency, I can revert this by using Jackson as a test dependency. But with our overall shift on limiting the usage of Jackson, I thought to remove using it as an example as well.

File: fluentgen/pom.xml (hunk 5 above)
Comment (alzimmermsft) [reply fluentgen/pom.xml:70]: Needed as fluentgen has tests that use Azure SDK types that still depend on Jackson.

File: javagen/src/main/java/com/azure/autorest/util/TemplateUtil.java (hunk 2 above)
Comment (alzimmermsft) [reply javagen/src/main/java/com/azure/autorest/util/TemplateUtil.java:81]: I haven't run across issues with this being removed. Unless this is something I need to regenerate an actual Azure SDK with to verify.

File: customization-base/src/main/java/com/azure/autorest/customization/implementation/ls/EclipseLanguageClient.java (hunk 1 above)
Comment (srnagar) [reply customization-base/src/main/java/com/azure/autorest/customization/implementation/ls/EclipseLanguageClient.java:212]: It doesn't have to be done in this PR. Please file an issue and resolve this comment.

**Review by srnagar — APPROVED** [review]

**Pull request merged**